File: frontend/src/types/market.ts
// Mirrors backend schemas/market.py

export interface QuoteResponse {
  symbol: string;
  price: number;
  timestamp: string;
}

export interface AssetEntry {
  symbol: string;
  name: string;
}

export interface HistoricalBarResponse {
  timestamp: string;
  close: number;
}

File: backend/app/schemas/market.py
"""Schemas for market data endpoints exposed to the frontend."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class QuoteResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    symbol: str
    price: float
    timestamp: datetime
    
class HistoricalBarResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    timestamp: datetime
    close: float